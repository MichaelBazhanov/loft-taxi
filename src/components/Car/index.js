export { default } from "./Car";
export { styles } from "./styles";
