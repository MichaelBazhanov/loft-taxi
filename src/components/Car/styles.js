export const styles = {
  container: ({ flag }) =>
    ["w-1/3", , "cursor-pointer", flag ? "" : "opacity-50"].join(" "),
  car: ["flex", "flex-col", "shadow-me", "rounded-lg", "p-3"].join(" "),
  img: ["mx-auto", "mt-1", "object-cover", "object-center", "h-16"].join(" "),
  "price-name": ["text-xs"].join(" "),
  price: ["text-2xl"].join(" "),
};
