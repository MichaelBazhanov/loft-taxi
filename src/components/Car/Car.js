import React from "react";
import PropTypes from "prop-types";
import { styles } from "./styles";

const Cars = ({
  name,
  price,
  imgSrc,
  index,
  activeIndexCar,
  setActiveIndexCar,
}) => {
  return (
    <div
      className={styles.container({ flag: index === activeIndexCar })}
      onClick={() => setActiveIndexCar(index)}
    >
      <div className={styles.car}>
        <p>{name}</p>
        <p className={styles["price-name"]}>Стоимость</p>
        <p className={styles.price}>{price}</p>
        <img src={imgSrc} alt={name} className={styles.img} />
      </div>
    </div>
  );
};

Cars.propTypes = {
  price: PropTypes.string,
  imgSrc: PropTypes.string,
  index: PropTypes.number,
  activeIndexCar: PropTypes.number,
  setActiveIndexCar: PropTypes.func,
};

export default Cars;
