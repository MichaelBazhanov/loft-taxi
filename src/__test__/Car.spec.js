import React from "react";
import TestRenderer from "react-test-renderer";

import Car from "../components/Car";

it("Renders correctly", () => {
  const CarForFormExample = TestRenderer.create(<Car />).toJSON();

  expect(CarForFormExample).toMatchSnapshot();
});
