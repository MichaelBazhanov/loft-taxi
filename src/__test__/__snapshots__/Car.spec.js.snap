// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="w-1/3  cursor-pointer "
  onClick={[Function]}
>
  <div
    className="flex flex-col shadow-me rounded-lg p-3"
  >
    <p />
    <p
      className="text-xs"
    >
      Стоимость
    </p>
    <p
      className="text-2xl"
    />
    <img
      className="mx-auto mt-1 object-cover object-center h-16"
    />
  </div>
</div>
`;
